scanner/src/main/kotlin/scanners/scanoss/ScanOssConfig.kt

```diff
@@ -0,0 +1,95 @@
+
+import javax.net.ssl.HostnameVerifier
+import javax.net.ssl.SSLContext
+import javax.net.ssl.TrustManager
+import javax.net.ssl.X509TrustManager
+
+import okhttp3.OkHttpClient
+
+import org.ossreviewtoolkit.clients.fossid.FossIdServiceWithVersion
+import org.ossreviewtoolkit.clients.scanoss.ScanossService
+import org.ossreviewtoolkit.model.config.ScannerConfiguration
+import org.ossreviewtoolkit.utils.core.OkHttpClientHelper
+import org.ossreviewtoolkit.utils.core.log
+
+/**
+ * A data class that holds the configuration options supported by the [ScanOss] scanner. An instance of this class is
+ * created from the options contained in a [ScannerConfiguration] object under the key _ScanOss_. It offers the
+ * following configuration options:
+ */
+internal data class ScanOssConfig(
+    /** URL of the ScanOSS API server. */
+    val apiURL: String,
+
+    /** API Key required to authenticate with the ScanOSS API server. */
+    val apiKey: String
+) {
+    companion object {
+        /** Name of the configuration property for the API URL. */
+        private const val API_URL_PROPERTY = "apiURL"
+
+        /** Name of the configuration property for the API key. */
+        private const val API_KEY_PROPERTY = "apiKey"
+
+        fun create(scannerConfig: ScannerConfiguration): ScanOssConfig {
+            val scanOssIdScannerOptions = scannerConfig.options?.get("ScanOss")
+
+            requireNotNull(scanOssIdScannerOptions) { "No ScanOSS Scanner configuration found." }
+
+            val apiURL = scanOssIdScannerOptions[API_URL_PROPERTY] ?: "https://osskb.org/api/"
+            val apiKey = scanOssIdScannerOptions[API_KEY_PROPERTY].orEmpty()
+
+            return ScanOssConfig(apiURL, apiKey)
+        }
+    }
+
+    /**
+     * Create the [FossIdServiceWithVersion] to interact with the FossID instance described by this configuration.
+     */
+    fun createService(): ScanossService {
+        log.info { "The ScanOSS API URL is $apiURL." }
+        fun OkHttpClient.Builder.ignoreAllSSLErrors(): OkHttpClient.Builder {
+            val naiveTrustManager = object : X509TrustManager {
+                override fun getAcceptedIssuers(): Array<X509Certificate> = arrayOf()
+                override fun checkClientTrusted(certs: Array<X509Certificate>, authType: String) = Unit
+                override fun checkServerTrusted(certs: Array<X509Certificate>, authType: String) = Unit
+            }
+
+            val insecureSocketFactory = SSLContext.getInstance("TLSv1.2").apply {
+                val trustAllCerts = arrayOf<TrustManager>(naiveTrustManager)
+                init(null, trustAllCerts, SecureRandom())
+            }.socketFactory
+
+            sslSocketFactory(insecureSocketFactory, naiveTrustManager)
+            hostnameVerifier(HostnameVerifier { _, _ -> true })
+            return this
+        }
+
+        return ScanossService.create(apiURL, OkHttpClientHelper.buildClient() {
+            ignoreAllSSLErrors()
```

Comment:
I believe we shouldn't do this unconditionally. Maybe also add a SCANOSS-specific option for this?

Reply:
It should not be done at all ! :D
I needed to intercept HTTPS calls with Wiremock.

Reply:
Oh! If this is only needed for testing, we should probably also only configure the client this way for testing... but I'll have a look, maybe having this option for production code is also valuable.

Reply:
You only need it at the start of the testing, we you record the requests by putting Wiremock between the client and the server.
After that, the Wiremock replays the server responses but the connection between the test and Wiremock is plain HTTP.
Therefore I removed this code.
If you need it again, I have a Gist :)